src/core/Akka.Streams/ActorMaterializer.cs

```diff
@@ -42,18 +44,18 @@ public static Config DefaultConfig()
             => DefaultMaterializerConfig;
 
         #region static
-        
+
         /// <summary>
         /// Injecting the top-level Materializer HOCON configuration over and over again is expensive, so we want to avoid
         /// doing it each time a materializer is instantiated. This flag will be set to true once the configuration has been
         /// injected the first time.
         /// </summary>
-        private static volatile bool _injectedConfig = false;
+        private static readonly ConcurrentDictionary<ActorSystem, bool> InjectedConfig = new();
```

Comment:
This would be unnecessary if the materializer were an `ActorSystemExtension`, but since it's not - have to do it this way.